fastlane/lib/fastlane/cli_tools_distributor.rb

```diff
@@ -102,6 +102,12 @@ def take_off
             # When we launch this feature, this should never be the case
             abort("#{tool_name} can't be called via `fastlane #{tool_name}`, run '#{tool_name}' directly instead".red)
           end
+         
+          # Some of the tools use other actions so need to load all
+          # actions before we start the tool generator
+          # Example: scan uses slack
```

Comment:
Are there any other cases? Will this slow down things? Will this cause us to load things twice?

Reply:
Yeah, this same question cross my mind too. My initial solution was to actually test if the actions got loaded first before this line - https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/slack.rb#L227. This "worked" but felt really weird to load an action even though we were already in an action.

The next higher up solution was require only the files I needed but the also felt weird and not very maintainable.

This approach of doing it at a higher level is a lot more maintainable and will work for other tools that do similar things (if there are any). I'm actually surprised that this wasn't caught sooner, TBH.

But the one-off action runs and `Fastfile` uses cases call this same method so it doesn't seem too bad to also call this for tools, IMO 🙃 Especially if it will fix current issues like this or prevent others in the future.